adala/runtimes/_openai.py

```diff
@@ -64,86 +54,53 @@ async def async_create_completion(
             messages[0]["content"] += system_prompt
 
     try:
-        async with semaphore, session.post(
-            DEFAULT_CREATE_COMPLETION_URL,
-            headers={"Authorization": f"Bearer {openai_api_key}"},
-            json={
-                "messages": messages,
-                "model": model,
-                "max_tokens": max_tokens,
-                "temperature": temperature,
-            },
-        ) as response:
-            response.raise_for_status()
-            response_json = await response.json()
-            completion_text = response_json["choices"][0]["message"]["content"]
-            return {
-                "text": completion_text,
-            }
-    except aiohttp.ClientResponseError as e:
-        # Handle HTTP errors
-        return {
-            "error": True,
-            "message": f"HTTP error: {e.status}",
-            "details": str(e),
-        }
-    except aiohttp.ClientError as e:
-        # Handle other aiohttp specific errors
-        return {
-            "error": True,
-            "message": "Client error",
-            "details": str(e),
-        }
-    except asyncio.TimeoutError as e:
-        # Handle timeout errors
+        completion = await client.chat.completions.create(
+            model=model,
+            messages=messages,
+            max_tokens=max_tokens,
+            temperature=temperature,
+        )
+        completion_text = completion.choices[0].message.content
         return {
-            "error": True,
-            "message": "Request timed out",
-            "details": str(e),
+            "text": completion_text,
+            "__adala_error__": False,
```

Comment:
nit: I think it's more common to denote private "attributes" in python with a single "__" before the name (as opposed to private/reserved methods with the "__" on both sides)

Reply:
done, good call